src/hotspot/share/opto/macro.cpp

```diff
@@ -2581,7 +2583,30 @@ bool PhaseMacroExpand::expand_macro_nodes() {
       expand_subtypecheck_node(n->as_SubTypeCheck());
       break;
     default:
-      assert(false, "unknown node type in macro list");
+      switch (n->Opcode()) {
+      case Op_ModD:
+      case Op_ModF: {
+        bool is_drem = n->Opcode() == Op_ModD;
+        CallNode* mod_macro = n->as_Call();
+        CallNode* call = new CallLeafNode(mod_macro->tf(),
+                                          is_drem ? CAST_FROM_FN_PTR(address, SharedRuntime::drem) : CAST_FROM_FN_PTR(address, SharedRuntime::frem),
```

Comment:
```suggestion
                                          is_drem ? CAST_FROM_FN_PTR(address, SharedRuntime::drem)
                                                  : CAST_FROM_FN_PTR(address, SharedRuntime::frem),
```
The line is a little long.